### PR TITLE
readds guardians to the uplink

### PR DESCRIPTION
**Pull request by ghost**
:cl: Armie
tweak: Holoparasites are once again in the uplink.
/:cl:

These are fun, they require cooperation between the user and the holo, and they're great for new players. New players have trouble getting good and having someone to teach them how to antagonize while helping them survive sounds like a blast.

I'm really sad we removed them in the first place, for silly reasons too.

### CODE REVIEW

**Comment by Hatterhat** [comment]
YES

**Comment by Togopalis** [comment]
If its for new players then dont let people with account age over 90 days buy it

:-)

**Comment by uraniummeltdown** [comment]
based

**Comment by epicmemename** [comment]
can you get them on discount as a traitor?

**Comment by Armhulen** [comment]
Probably but you can't have more than one stand

**Comment by Cyberboss** [comment]
Where's your changelog

Also https://atlantaned.space/newSS13tools/coderbus/gbp.php >:(

**Comment by Dorsisdwarf** [comment]
I note that they're back at the cheap price rather than the high price

**Comment by ExcessiveUseOfCobblestone** [comment]
They should be 20 TC since having a partner who is forever stealthed and always by your side is really REALLY strong. Also they can probably remove doors???

**Comment by Armhulen** [comment]
I'll raise to 18

**Comment by Armhulen** [comment]
@Cyberboss I'll make it up after this one!

**Comment by Armhulen** [comment]
also, just saying like 2 of my fixes were labelled as tweaks so they took points instead of giving me them soooooooooo just saying !

**Comment by MoreRobustThanYou** [comment]
@Armhulen we dont have gangs anymore, remove it from exclude

File: code/modules/uplink/uplink_item.dm [hunk 1]
```diff
@@ -368,6 +368,15 @@ GLOBAL_LIST_EMPTY(uplink_items) // Global list so we only initialize this once.
...
 	surplus = 35
 	include_modes = list(/datum/game_mode/nuclear)
+	
+/datum/uplink_item/dangerous/guardian
+	name = "Holoparasites"
+	desc = "Though capable of near sorcerous feats via use of hardlight holograms and nanomachines, they require an \
+			organic host as a home base and source of fuel."
+	item = /obj/item/weapon/storage/box/syndie_kit/guardian
+	cost = 18
+	exclude_modes = list(/datum/game_mode/nuclear, /datum/game_mode/gang)
```
Comment (KorPhaeron) [inline code/modules/uplink/uplink_item.dm]: >gang

Comment (Armhulen) [reply code/modules/uplink/uplink_item.dm]: this explains travis

**Comment by Repukan** [comment]
They were extremely unbalanced, hence why they were removed.

They outright made more than 60% of the purchases and if re added it will be less than two weeks before they are removed again.

File: code/modules/uplink/uplink_item.dm [hunk 2]
```diff
@@ -368,6 +368,15 @@ GLOBAL_LIST_EMPTY(uplink_items) // Global list so we only initialize this once.
 	cost = 12
 	surplus = 35
 	include_modes = list(/datum/game_mode/nuclear)
+	
+/datum/uplink_item/dangerous/guardian
+	name = "Holoparasites"
+	desc = "Though capable of near sorcerous feats via use of hardlight holograms and nanomachines, they require an \
+			organic host as a home base and source of fuel."
+	item = /obj/item/weapon/storage/box/syndie_kit/guardian
```
Comment (PKPenguin321) [inline code/modules/uplink/uplink_item.dm]: travis says this path doesnt exist

Comment (XDTM) [reply code/modules/uplink/uplink_item.dm]: weapon was removed, make it item/storage/...

**Comment by XDTM** [comment]
Holoparasites are way too powerful in the stun meta since they can't be stunned themselves, and can drag their stunned host to safety until it wears off, and this is before considering the wide variety of stats and powers they can have.

**Comment by AnonymousNow** [comment]
They'd be better on antagonist types that don't fire a massive ball of... well, fire at will, since they're less likely to accidentally kill their user.

They can be very appropriate on some wizard builds as well, but I like the idea of having them available for traitors.

**Comment by XDTM** [comment]
i feel like they'd have a nice niche if they were less combat punchghosts and more utility allies who can't physically interact outside of their unique abilities. (For example, the bomb one, or the scout mode of the ranged one)

**Comment by Repukan** [comment]
Don't pull a fucking pk and re add something that was broken and unbalanced because "it's fun lmao"

**Comment by uraniummeltdown** [comment]
yeah God forbid players have fun

**Comment by Repukan** [comment]
Fun was the reason they were removed you stupid cock whisperer

**Comment by uraniummeltdown** [comment]
fuck off you dont even contribute

**Comment by XDTM** [comment]
considering it takes a lot to remove features in this codebase, i think that there probably was a valid reason for their removal, readding them as they are is not gonna be good

**Comment by uraniummeltdown** [comment]
The PR removing them says they were removed for being "thematically inappropriate" nothing about balance or broken or OP

**Comment by XDTM** [comment]
There's also the whole backstory of them completely countering stun combat. 

You didn't powergame and bring a shotgun from the armory rounstart? Sucks for you and your taser/stunbaton.

**Comment by Repukan** [comment]
@KorPhaeron please weight on this.

**Comment by Armhulen** [comment]
@XDTM i did change it, I increased the price to sleeping carp.

**Comment by Armhulen** [comment]
it is up to kor though yeah, I just think he'll support new players and stuff

**Comment by Qbopper** [comment]
I love ~stands~ holoparasites so hopefully there can bea compromise when it comes to balance

**Comment by Bawhoppen** [comment]
What if we just make them stunnable in some special way?

File: code/modules/uplink/uplink_item.dm (hunk 2 above)
Comment (ShizCalev) [inline code/modules/uplink/uplink_item.dm]: item = /obj/item/storage/box/syndie_kit/guardian

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/modules/uplink/uplink_item.dm (hunk 2 above)
Comment (Armhulen) [reply code/modules/uplink/uplink_item.dm]: thank you, and i will do!

**Review by ShizCalev — APPROVED** [review]

**Comment by KorPhaeron** [comment]
@Repukan SS13 is cyclical. Each generation of newfriends has to learn the same lessons again

**Comment by Armhulen** [comment]
@Bawhoppen i wouldn't be against moving stuns on the holo to the master but yknow

for 18 tc (more than sleeping carp) i think this is fair

**Comment by zamolxius** [comment]
yes please

**Comment by ChangelingRain** [comment]
wasn't there something about leaving a comment when closing prs?

**Comment by Armhulen** [comment]
oranges hit and run

**Comment by tgstation-server** [comment]
Your Fix/Feature pull request delta is currently below zero (-3). Maintainers may close future Feature/Tweak/Balance PRs. Fixing issues or helping to improve the codebase will raise this score.

**Comment by optimumtact** [comment]
yeah these were removed because of how powerful it is, @KorPhaeron's comment is indicative of that.

There's no reason to go through it again.

**Comment by GuyonBroadway** [comment]
Hippie has had these for months. You hardly see them which is sad.

**Comment by Bawhoppen** [comment]
Hippie doesn't powergame nearly as much

**Comment by uraniummeltdown** [comment]
How is a traitor using an item in his uplink powergaming?

**Pull request merged**